src/sys/ptrace.rs

```diff
@@ -70,7 +70,7 @@ mod ffi {
 
 /// Performs a ptrace request. If the request in question is provided by a specialised function
 /// this function will return an unsupported operation error.
-pub fn ptrace(request: ptrace::PtraceRequest, pid: Pid, addr: *mut c_void, data: *mut c_void) -> Result<c_long> {
+pub unsafe fn ptrace(request: ptrace::PtraceRequest, pid: Pid, addr: *mut c_void, data: *mut c_void) -> Result<c_long> {
```

Comment:
And thus break the wrapper: `ptrace::syscall` calls `ptrace::ptrace` with `PTRACE_SYSCALL`.

`ptrace` can dereference arbitrary pointer in the child process memory.